packages/widget/src/stores/bookmarks/createBookmarkStore.ts

```diff
@@ -57,6 +63,16 @@ export const createBookmarksStore = ({ namePrefix }: PersistStoreProps) =>
           bookmarks: state.bookmarks,
           recentWallets: state.recentWallets,
         }),
+        onRehydrateStorage: () => {
+          return (state) => {
+            if (state && toAddress && !toAddress.name) {
+              const existingBookmark = state.getBookmark(toAddress.address);
+              if (existingBookmark) {
+                state.setSelectedBookmark(existingBookmark);
+              }
+            }
+          };
```

Comment:
Yesterday we tried and seemed to work with the main function, not return one 👀 Did you find something?

Reply:
I double checked it. The main function doesn't have the bookmarks from local storage  - the bookmarks are only populated in the returned function. Might have been outputting the wrong things in the console yesterday?? When moving the functionality to the returned function it works 👍